e2e-app/src/app/typeahead/focus/typeahead-focus.component.ts

```diff
@@ -1,86 +1,43 @@
-import {Component, ViewChild} from '@angular/core';
+import {HttpClient, HttpParams} from '@angular/common/http';
+import {Component, ViewChild, Injectable} from '@angular/core';
 import {NgbTypeahead} from '@ng-bootstrap/ng-bootstrap';
 import {Observable, Subject, merge} from 'rxjs';
-import {distinctUntilChanged, filter, map} from 'rxjs/operators';
+import {debounceTime, distinctUntilChanged, filter, map, tap, switchMap} from 'rxjs/operators';
 
-const states = [
-  'Alabama',
-  'Alaska',
-  'American Samoa',
-  'Arizona',
-  'Arkansas',
-  'California',
-  'Colorado',
-  'Connecticut',
-  'Delaware',
-  'District Of Columbia',
-  'Federated States Of Micronesia',
-  'Florida',
-  'Georgia',
-  'Guam',
-  'Hawaii',
-  'Idaho',
-  'Illinois',
-  'Indiana',
-  'Iowa',
-  'Kansas',
-  'Kentucky',
-  'Louisiana',
-  'Maine',
-  'Marshall Islands',
-  'Maryland',
-  'Massachusetts',
-  'Michigan',
-  'Minnesota',
-  'Mississippi',
-  'Missouri',
-  'Montana',
-  'Nebraska',
-  'Nevada',
-  'New Hampshire',
-  'New Jersey',
-  'New Mexico',
-  'New York',
-  'North Carolina',
-  'North Dakota',
-  'Northern Mariana Islands',
-  'Ohio',
-  'Oklahoma',
-  'Oregon',
-  'Palau',
-  'Pennsylvania',
-  'Puerto Rico',
-  'Rhode Island',
-  'South Carolina',
-  'South Dakota',
-  'Tennessee',
-  'Texas',
-  'Utah',
-  'Vermont',
-  'Virgin Islands',
-  'Virginia',
-  'Washington',
-  'West Virginia',
-  'Wisconsin',
-  'Wyoming'
-];
+const DATA_URL = '/public/typeahead.json';
+const PARAMS = new HttpParams({});
 
-@Component({templateUrl: './typeahead-focus.component.html'})
+@Injectable()
+export class SearchService {
+  constructor(private http: HttpClient) {}
+
+  search(term: string) {
+    return this.http.get(DATA_URL, {params: PARAMS.set('search', term)}).pipe(map((response: any) => {
+      return response.filter(v => v.toLowerCase().indexOf(term.toLowerCase()) > -1).slice(0, 10);
+    }));
+  }
+}
+
+
+@Component({templateUrl: './typeahead-focus.component.html', providers: [SearchService]})
 export class TypeaheadFocusComponent {
   model: any;
+  searching = false;
+  searchFailed = false;
 
   @ViewChild('instance') instance: NgbTypeahead;
   focus$ = new Subject<string>();
   click$ = new Subject<string>();
 
+  constructor(private _service: SearchService) {}
+
   search = (text$: Observable<string>) => {
-    const debouncedText$ = text$.pipe(distinctUntilChanged());
+
+    const debouncedText$ = text$.pipe(debounceTime(200), distinctUntilChanged());
     const clicksWithClosedPopup$ = this.click$.pipe(filter(() => !this.instance.isPopupOpen()));
     const inputFocus$ = this.focus$;
 
     return merge(debouncedText$, inputFocus$, clicksWithClosedPopup$)
```

Comment:
I think the problem is that current code is synchronous. To reproduce we should just add the `delay(0)` operator, maybe ?

```ts
return merge(...)
    .pipe(
        map(...),
        delay(0)
    );
```

And remove all the changes with HTTP to keep use cases as simple as possible